src/test/java/codegym/tequila/fisioapp/service/impl/MedicalRecordServiceImplTest.java

```diff
@@ -0,0 +1,232 @@
+        assertThat(medicalRecordSaved.getPersonalRecords().getMusculoskeletalConditions()).isEqualTo(medicalRecordDto.getPersonalRecords().getMusculoskeletalConditions());
+        assertThat(medicalRecordSaved.getFamiliarRecords().getGeneticalPredispositions()).isEqualTo(medicalRecordDto.getFamiliarRecords().getGeneticalPredispositions());
+        assertThat(medicalRecordSaved.getFamiliarRecords().getHereditaryDiseases()).isEqualTo(medicalRecordDto.getFamiliarRecords().getHereditaryDiseases());
+
+        assertThat(medicalRecordSaved.getPatient()).withFailMessage("Patient can not be changed").isEqualTo(medicalRecordDao.getPatient());
+
+        medicalRecordDto.setId(medicalRecordDtoReturned.getId());
+        assertThat(medicalRecordDto).isEqualTo(medicalRecordDtoReturned);
+
+        verify(medicalRecordRepository).save(medicalRecordSaved);
+        verify(medicalRecordRepository).findById(medicalRecordDto.getId());
+        verifyNoMoreInteractions(medicalRecordRepository);
+    }
+
+    @Test
+    public void updateMedicalRecordWithInvalidIdTest() {
+        //Given:
+        medicalRecordDto.setId(UUID.randomUUID().toString());
+
+        when(medicalRecordRepository.findById(medicalRecordDto.getId())).thenReturn(Optional.empty());
+
+        //When:
+        assertThatThrownBy(() -> medicalRecordService.updateMedicalRecord(medicalRecordDto))
+                .isInstanceOf(NoSuchElementException.class)
+                .hasMessage("Medical Record " + medicalRecordDto.getId() + " not found");
+        //Then:
+        verify(medicalRecordRepository).findById(medicalRecordDto.getId());
+        verifyNoMoreInteractions(medicalRecordRepository);
+
+    }
+
+    @Test
+    public void findByIdTest() {
+        //Given:
+        MedicalRecord medicalRecordDao = new MedicalRecord();
+        String id = UUID.randomUUID().toString();
+
+        medicalRecordDao.setId(id);
+        medicalRecordDao.setPhysicalExploration(MedicalRecordServiceImpl.convertPhysicalExplorationDtoToPhysicalExploration(medicalRecordDto.getPhysicalExploration()));
+        medicalRecordDao.setPatient(MedicalRecordServiceImpl.convertPatientDtoToPatient(medicalRecordDto.getPatient()));
+        medicalRecordDao.setPersonalRecords(MedicalRecordServiceImpl.convertPersonalRecordsDtoToPersonalRecords(medicalRecordDto.getPersonalRecords()));
+        medicalRecordDao.setFamiliarRecords(MedicalRecordServiceImpl.convertFamiliarRecordsDtoToFamiliarRecords(medicalRecordDto.getFamiliarRecords()));
+
+
+        when(medicalRecordRepository.findById(id)).thenReturn(Optional.of(medicalRecordDao));
+
+        //When:
+        MedicalRecordDto medicalRecordDtoResult = medicalRecordService.findById(id);
+
+        //Then:
+        assertThat(medicalRecordDtoResult).isNotNull();
+        assertThat(medicalRecordDtoResult.getId()).isEqualTo(id);
+        assertThat(medicalRecordDtoResult.getPhysicalExploration()).isEqualTo(medicalRecordDto.getPhysicalExploration());
```

Comment:
Compare against the database value which is represented by your object `medicalRecordDao`